appsmithorg/appsmith
# feat: Deprecate form button widget

## Description
Since we have `resetWidget` as a trigger function, we can deprecate the form button widget added with every form widget. 


- [ ] Rename all references of `FORM_BUTTON_WIDGET` to `BUTTON_WIDGET`
- [ ] Remove `FormButtonWidget` from the widgets in Appsmith
- [ ] Add migrations to change the type of `FORM_BUTTON_WIDGET` to `BUTTON_WIDGET`
- [ ] Migrate the extra features existing in FormButtonWidget to ButtonWidget.

## QA Note: 
This needs to be a backwards compatible change, i.e, existing Form Widgets must work as they have been so far.
We need to also confirm if FormButtonWidget's features work in the ButtonWidget now. (Reset on click and Disabled on invalid)

Fixes #3613 

## Type of change

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How Has This Been Tested?

- Migrations have jest tests
- Existing Form Widget tests must pass.

## Checklist:








## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>


    // Code coverage diff between base branch:release and head branch: feat/3613-deprecate-form-button 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.71 **(-0.01)** | 37.09 **(0.04)** | 35.79 **(0)** | 56.07 **(-0.01)**
 :green_circle: | app/client/src/icons/WidgetIcons.tsx | 59.78 **(0.64)** | 100 **(0)** | 15.91 **(0.35)** | 59.78 **(0.64)**
 :green_circle: | app/client/src/sagas/SnipingModeSagas.ts | 20 **(0)** | 3.23 **(0.1)** | 40 **(0)** | 20.79 **(0)**
 :green_circle: | app/client/src/utils/DSLMigrations.ts | 73.63 **(0.36)** | 70.05 **(0.72)** | 71.72 **(0.59)** | 73.46 **(0.38)**
 :green_circle: | app/client/src/utils/WorkerUtil.ts | 89.76 **(0.78)** | 72.55 **(1.96)** | 100 **(0)** | 93.33 **(0.95)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :red_circle: | app/client/src/widgets/ButtonWidget/widget/index.tsx | 61.22 **(-9.51)** | 30 **(-12.86)** | 58.33 **(0)** | 60.42 **(-9.58)**
 :x: | ~~app/client/src/widgets/FormButtonWidget/index.ts~~ | ~~100~~ | ~~100~~ | ~~100~~ | ~~100~~
 :x: | ~~app/client/src/widgets/FormButtonWidget/widget/index.tsx~~ | ~~65.71~~ | ~~19.05~~ | ~~62.5~~ | ~~64.71~~
 :x: | ~~app/client/test/factories/Widgets/FormButtonFactory.ts~~ | ~~75~~ | ~~100~~ | ~~0~~ | ~~75~~</details>
